ui/src/components/CheckboxTree/CheckboxSubTree.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import CheckboxRowWrapper from './CheckboxTreeRowWrapper';
+import { getCheckedCheckboxesCount } from './CheckboxTree.utils';
+import {
+    CheckboxContainer,
+    CheckboxWrapper,
+    CustomCheckbox,
+    Description,
+    GroupLabel,
+    RowContainer,
+    StyledCollapsiblePanel,
+} from './StyledComponent';
+import { GroupWithRows, ValueByField } from './types';
+
+interface CheckboxSubTreeProps {
+    group: GroupWithRows;
+    values: ValueByField;
+    handleRowChange: (newValue: { field: string; checkbox: boolean; text?: string }) => void;
+    disabled?: boolean;
+    handleParentCheckboxForGroup: (groupLabel: string, newCheckboxValue: boolean) => void;
+}
+
+const CheckboxSubTree: React.FC<CheckboxSubTreeProps> = ({
+    group,
+    values,
+    handleRowChange,
+    disabled,
+    handleParentCheckboxForGroup,
+}) => {
+    const [isExpanded, setIsExpanded] = useState(group.options?.expand);
+
+    const isParentChecked = useMemo(
+        () => group.rows.every((row) => values.get(row.field)?.checkbox),
+        [group.rows, values]
+    );
+
+    const isIndeterminate = useMemo(
+        () => !isParentChecked && group.rows.some((row) => values.get(row.field)?.checkbox),
+        [group.rows, values, isParentChecked]
+    );
+
+    const checkedCheckboxesCount = useMemo(
+        () => getCheckedCheckboxesCount(group, values),
+        [group, values]
+    );
+
+    useEffect(() => {
+        setIsExpanded(group.options?.expand);
+    }, [group.options?.expand, group.rows]);
+
+    const toggleCollapse = () => setIsExpanded((prev) => !prev);
+
+    const ParentCheckbox = (
+        <CheckboxWrapper>
+            <CustomCheckbox
+                type="checkbox"
+                checked={isParentChecked}
+                data-indeterminate={isIndeterminate}
+                onChange={() => handleParentCheckboxForGroup(group.label, !isParentChecked)}
+                disabled={disabled || group.options?.disabled}
+                aria-label="custom checkbox for group"
```

Comment:
The label is not correct since it does not reflect the real meaning for that checkbox

Reply:
In this, we have 3 state:
- select all
- deselect all
- indeterminate

but splunk don't have support for inderminate, that's why I have added a custom checkbox so that we can use the upper 3 states for checkbox